src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,267 @@
+import {
+  allowedMasks
+} from '../util/mask'
+
+export default {
+  data: () => ({
+    options: {},
+    signSymbol: '',
+    preDefined: {
+      'numeral': {
+        positiveOnly: false, // Positive value only
+        positiveSign: false, // Show positive sign '+'
+        blockSeparator: ' ', // {String|Array} - Separate between groups of digits
+        blockSize: 3, // {Number|Array} - Number of digits per group
+        decimal: '.', // Decimal mark
+        prefix: '', // Prefix shown in input field
+        precision: 2, // Floating point precision
+        suffix: '', // Suffix shown in input field
+        formatter: 'numeral'
+      }
+    }
+  }),
+
+  computed: {
+    precision () {
+      return this.absVal(this.options.precision)
+    },
+    positiveSign () {
+      return this.options.positiveSign ? '+' : ''
+    },
+    decimal () {
+      return this.oneChar(this.options.decimal)
+    },
+    blockSeparator () { // Default separator = ' '
+      const separator = this.options.blockSeparator
+
+      if (!Array.isArray(separator)) return [this.oneChar(separator, '')]
+
+      switch (separator.length) {
+        case 0: return [' ']
+        case 1: return this.oneChar(separator, '')
+        default: return separator
+      }
+    },
+    blockSize () { // Default size = 3
+      const size = this.options.blockSize
+
+      if (!Array.isArray(size)) return [this.absVal(size)]
+
+      switch (size.length) {
+        case 0: return [3]
+        case 1: return [this.absVal(size[0])]
+        default: return size
+      }
+    },
+    sign: {
+      get () {
+        return this.signSymbol
+      },
+      set (val) {
+        let i = 0
+
+        if (this.numeralPrefix) {
+          for (const char of val) {
+            if (allowedMasks['#'].test(char)) break
+            if (char === '+' || char === '-') break
+            i++
+          }
+          val = val.substr(i)
+        }
+
+        const negative = val && val.length > 0 && val[0] === '-'
+        const positiveSymbol = this.positiveSign ? '+' : ''
+
+        this.signSymbol = this.options.positiveOnly ? positiveSymbol
+          : negative ? '-' : positiveSymbol
+      }
+    },
+    numeralPrefix () {
+      return this.options.prefix ? this.options.prefix : ''
+    },
+    numeralSuffix () {
+      return this.options.suffix ? this.options.suffix : ''
+    }
+  },
+
+  methods: {
+    unmaskNumeralText (text) {
+      if (!text) return text
+
+      this.sign = text = String(text)
+      text = this.internalChange ? this.removeNonNumeral(text) : this.attemptNumeralCorrection(text)
+
+      return this.parseNumber(text)
+    },
+    maskNumeralText (text) {
+      if (!text) return this.numeralZero()
+      text = this.sign ? String(text).substr(1) : String(text)
+
+      const param = {
+        masked: [],
+        i: this.blockSeparator.length - 1,
```

Comment:
`i`, `j` are a bit mysterious names, maybe `blockNumber`, `charInBlock` (or whatever is these vars meaning)?

Reply:
Blame my C background